AppCenterAuth/AppCenterAuth/MSAuth.m

```diff
@@ -361,67 +376,80 @@ - (BOOL)removeAccount {
+                      return;
+                    }
+                    if (error) {
+                      NSString *errorMessage =
+                          [NSString stringWithFormat:@"Silent acquisition of token failed with error: %@.", error.localizedDescription];
+                      if ([error.domain isEqual:MSALErrorDomain] && error.code == MSALErrorInteractionRequired) {
+                        if (uiFallback) {
+                          MSLogInfo([MSAuth logTag], @"%@ Triggering interactive acquisition.", errorMessage);
+                          [strongSelf acquireTokenInteractivelyWithKeyPathForCompletionHandler:completionHandlerKeyPath];
+                          return;
+                        } else {
+                          MSLogError([MSAuth logTag], @"%@ But interactive acquisition fallback is not allowed here.", errorMessage);
+                        }
+                      } else {
+                        MSLogError([MSAuth logTag], @"%@", errorMessage);
+                      }
+                      [[MSAuthTokenContext sharedInstance] setAuthToken:nil withAccountId:nil expiresOn:nil];
+                      handler(nil, error);
                     } else {
                       MSALAccountId *accountId = (MSALAccountId * __nonnull) result.account.homeAccountId;
                       [[MSAuthTokenContext sharedInstance] setAuthToken:result.idToken
                                                           withAccountId:accountId.identifier
                                                               expiresOn:result.expiresOn];
-                      [strongSelf completeAcquireTokenRequestForResult:result withError:nil];
                       MSLogInfo([MSAuth logTag], @"Silent acquisition of token succeeded.");
+                      MSUserInformation *userInformation = [MSUserInformation new];
+                      userInformation.accountId = (NSString * __nonnull) result.uniqueId;
+                      handler(userInformation, nil);
                     }
                   }];
 }
 
-- (void)acquireTokenInteractively {
+- (void)acquireTokenInteractivelyWithKeyPathForCompletionHandler:(NSString *)completionHandlerKeyPath {
   __weak typeof(self) weakSelf = self;
   [self.clientApplication acquireTokenForScopes:@[ (NSString * __nonnull) self.authConfig.authScope ]
-                                completionBlock:^(MSALResult *result, NSError *e) {
+                                completionBlock:^(MSALResult *result, NSError *error) {
                                   typeof(self) strongSelf = weakSelf;
-                                  if (e) {
+                                  MSSignInCompletionHandler handler = [strongSelf valueForKey:completionHandlerKeyPath];
+                                  if (!handler) {
+                                    MSLogDebug([MSAuth logTag], @"Sign-in has been interrupted. Ignoring the result.");
```

Comment:
We don't have this case. Multiple signIn requests will ignore a new request.

Reply:
We **have** this one because of disabling or sign out for example